packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1126,10 +1122,6 @@ export class ProjectLifecycleManager {
       hasCypressEnvFile: fs.existsSync(this._pathToFile('cypress.env.json')),
     }
 
-    if (configFile === false) {
-      return metaState
-    }
-
```

Comment:
below on line 1151, do we need the `if (typeof configFile === 'string') {` check?

Reply:
Good catch, we can just change that to check for a truthy value. It can be null here since this is where the default gets detected if none is specified.

Reply:
Done in [cd40998](https://github.com/cypress-io/cypress/pull/20643/commits/cd40998ed0648f89da31ba9cf652684efde28713)